src/sagas/networkSettings.js

```diff
@@ -217,31 +238,42 @@ function invalidUrl(tryUrl) {
 export function* persistNetworkSettings(action) {
   // persists after reducer being updated
   const networkSettings = action.payload;
-  const strNetworkSettings = JSON.stringify(networkSettings);
-  yield call(AsyncStorage.setItem, networkSettingsKeyMap.networkSettings, strNetworkSettings);
-
-  // trigger toggle update to be managed by featureToggle saga
-  yield put(featureToggleUpdate());
-
-  // if wallet-service is being deactivated, it will trigger the reload,
-  // otherwise we should trigger by ourselves
-  const { timeout } = yield race({
-    reload: take(types.RELOAD_WALLET_REQUESTED),
-    timeout: delay(HTTP_REQUEST_TIMEOUT),
-  });
+  try {
+    STORE.setItem(networkSettingsKeyMap.networkSettings, networkSettings);
+    yield put(networkSettingsUpdateWaiting());
+  } catch (err) {
+    console.error('Error while persisting the custom network settings.', err);
+    yield put(networkSettingsUpdateFailure());
+    return;
+  }
 
-  if (timeout) {
-    yield put(reloadWalletRequested());
+  const wallet = yield select((state) => state.wallet);
+  if (!wallet) {
+    // If we fall into this situation, the app should be killed
+    // for the custom new network settings take effect.
+    const errMsg = 'Wallet not found while trying to persist the custom network settings.';
+    console.warn(errMsg);
+    yield put(onExceptionCaptured(errMsg, /* isFatal */ true));
+    return;
   }
 
-  yield put(networkSettingsUpdateReady());
```

Comment:
Is the state going to be waiting forever? When does it become READY?

Reply:
When the wallet reloads after network settings persistence, the `initNetworkSettings` effect is triggered, and this will change the status to `SUCCESSFUL`, which will show up the success feedback to the user. On the success dismiss, the status is changed to `READY`.

But I realize now that if something goes wrong I need a fallback, right? I will implement this fallback in the `initNetworkSettings` effect.

Reply:
Add fallback: 393eb4be8cda595460ada6bc15bdfdfe037368ac